pkg/test/environment.go

```diff
@@ -123,6 +123,13 @@ func NewEnvironment(scheme *runtime.Scheme, options ...functional.Option[Environ
 	} else {
 		c = lo.Must(client.New(environment.Config, client.Options{Scheme: scheme}))
 	}
+
+	namespace := corev1.Namespace{}
```

Comment:
Does this work all of the time? Can we try doing this with a continual retry for up to say 10s with a 1s back-off? Consider `retry.Do()`? The other thing: I think that we should add a comment here with a `// TODO` to remove this once we upgrade to the latest version of controller-runtime